README.md

```diff
@@ -1 +1,72 @@
-# noir_backend_using_gnark
\ No newline at end of file
+# Using gnark as a backend for Noir
+
+If you want to develop a backend for Noir, basically what you need to do is to translate ACIR (Abstract Circuit Intermediate Representation) to something that your backend could understand.
+The prerequisites for this are to have certain knowledge about ACIR and its codebase and to have certain knowledge about the backend that you are going to support.
+In this case, the backend is [gnark](https://github.com/ConsenSys/gnark). Written in Go, gnark is a fast zk-SNARK library that offers a high-level API to design circuits. The library is open-source and developed under the Apache 2.0 license 
+
+## Overview
+
+```mermaid
+flowchart
+    N[Noir] --ACIR--> BR[Backend in gnark wrapper in Rust]
+    BR[Backend in gnark wrapper written in Rust] --Raw R1CS--> B[Concrete Backend]
+    subgraph B[Concrete Backend]
+        direction TB
+        BW[WASM/FFI] --> BG[Bakend in gnark written in Go]
+    end
+```
+
+### Backend wrapper written in Rust
+
+This module is needed because a Noir's backend has to be written in Rust and we want to develop a backend using gnark which is written in Go.
```

Comment:
```suggestion
This module is needed because Noir's backend has to be written in Rust and we want to develop one using gnark which is written in Go.
```